users/views.py

```diff
@@ -210,3 +211,41 @@ def get_context_data(self, **kwargs: Any) -> dict:
         context = super().get_context_data(**kwargs)
         context["is_activated"] = self.activate()
         return context
+
+
+@method_decorator(login_required, name="dispatch")
+@method_decorator(
+    check_permissions(allowed_user_types=[CustomUser.UserType.ADMIN.name, CustomUser.UserType.MANAGER.name]),
+    name="dispatch",
+)
+class NotificationUserListView(ListView):
+    template_name = "user_notifications_list.html"
+    model = CustomUser
+
+    def get_queryset(self) -> QuerySet:
+        return CustomUser.objects.filter(projects__managers__pk=self.request.user.pk).distinct()
+
+    def get_context_data(self, *, _object_list: Any = None, **kwargs: Any) -> dict:
+        context_data = super().get_context_data(**kwargs)
```

Comment:
As discussed, I think this can be done with single queryset.

Reply:
Done

Reply:
Then you don't need the whole calculation below this comment. You already have date of last report for every user. You can easily calculate the difference between today and that date. In fact, when I was writing about doing this with single queryset, I have meant that it can be done in that queryset too using `F` object. Rough example:

```
for x in CustomUser.objects.annotate(last_report_date=Coalesce(Max("report__date"), None)).annotate(z=timezone.now() - F('last_report_date')): 
    ...:     print (x.z) 
    ...:                                                                                                                                                 
None
10 days, 21:49:57.549116
43 days, 21:49:57.549116
19 days, 21:49:57.549116
43 days, 21:49:57.549116
43 days, 21:49:57.549116
800 days, 21:49:57.549116
None
None
```

The only thing missing is skipping weekends but that can be added as simple template tag called `substract_weekdays_amount_from_last_days(days)` where `days` will be int taken from `z`.

Btw. I think that weekdays should count too, as the result might be ambiguous, for example there will be 4 days which can mean that there were really 4 days, or 6 but there was a weekend in meantime. Also, it is easier to notice that someone didn't report for example for whole 14 days but it is known he was on 2 weeks holidays, than to see 14 days without being sure if that is more than 2 weeks because weekends are cut off.

Reply:
Done. I left skipping weekends as request from client.